### PR TITLE
fixed bugs and added AI mushrooms (made a prefab)

### PR DESCRIPTION
**Pull request by benny-n**

### CODE REVIEW

File: The Middle Ages/Assets/Game/Prefabs/Core.prefab [hunk 1]
```diff
@@ -1939,7 +1939,7 @@ MonoBehaviour:
   m_OnClick:
     m_PersistentCalls:
       m_Calls:
-      - m_Target: {fileID: 0}
+      - m_Target: {fileID: 4000551786404188881}
```
Comment (orenkoren) [inline The Middle Ages/Assets/Game/Prefabs/Core.prefab:1942]: what is this?

File: The Middle Ages/Assets/Game/Prefabs/Core.prefab [hunk 2]
```diff
@@ -2019,8 +2019,8 @@ MonoBehaviour:
   m_OnCullStateChanged:
     m_PersistentCalls:
       m_Calls: []
-  m_Sprite: {fileID: 10907, guid: 0000000000000000f000000000000000, type: 0}
-  m_Type: 1
+  m_Sprite: {fileID: 21300000, guid: 9d5f6625b0b035841a1153f7e5466a94, type: 3}
```
Comment (orenkoren) [inline The Middle Ages/Assets/Game/Prefabs/Core.prefab:2022]: what is this?

File: The Middle Ages/Assets/Game/Prefabs/Core.prefab [hunk 3]
```diff
@@ -1506,8 +1506,8 @@ MonoBehaviour:
   m_faceColor:
     serializedVersion: 2
     rgba: 4294967295
-  m_fontSize: 15
-  m_fontSizeBase: 15
+  m_fontSize: 10.19
```
Comment (orenkoren) [inline The Middle Ages/Assets/Game/Prefabs/Core.prefab:1509]: what is this?

File: The Middle Ages/Assets/Game/Prefabs/Core.prefab [hunk 4]
```diff
@@ -1448,9 +1448,9 @@ RectTransform:
   m_LocalEulerAnglesHint: {x: 0, y: 0, z: 0}
   m_AnchorMin: {x: 0.5, y: 0.5}
   m_AnchorMax: {x: 0.5, y: 0.5}
-  m_AnchoredPosition: {x: 65.9, y: -6.4}
+  m_AnchoredPosition: {x: 74.3, y: -10.2}
```
Comment (orenkoren) [inline The Middle Ages/Assets/Game/Prefabs/Core.prefab:1451]: what is this?

File: The Middle Ages/Assets/Game/Scripts/UI Canvas/ExpBarHandler.cs [hunk 5]
```diff
@@ -9,5 +9,11 @@ void Start()
         {
             resource = ((PlayerStats)stats).GetExperiencePoints();
         }
+
+        protected override void UpdateResourceBarAccordingToCurrentResourcePoints(int currentRp)
```
Comment (orenkoren) [inline The Middle Ages/Assets/Game/Scripts/UI Canvas/ExpBarHandler.cs:13]: what is this?

File: The Middle Ages/Assets/Game/Prefabs/Core.prefab (hunk 1 above)
Comment (benny-n) [reply The Middle Ages/Assets/Game/Prefabs/Core.prefab:1942]: related to the "exit to main menu" button bug fix

File: The Middle Ages/Assets/Game/Prefabs/Core.prefab (hunk 2 above)
Comment (benny-n) [reply The Middle Ages/Assets/Game/Prefabs/Core.prefab:2022]: made exp bar look better

File: The Middle Ages/Assets/Game/Prefabs/Core.prefab (hunk 3 above)
Comment (benny-n) [reply The Middle Ages/Assets/Game/Prefabs/Core.prefab:1509]: same as above

File: The Middle Ages/Assets/Game/Prefabs/Core.prefab (hunk 4 above)
Comment (benny-n) [reply The Middle Ages/Assets/Game/Prefabs/Core.prefab:1451]: same as above

File: The Middle Ages/Assets/Game/Scripts/UI Canvas/ExpBarHandler.cs (hunk 5 above)
Comment (benny-n) [reply The Middle Ages/Assets/Game/Scripts/UI Canvas/ExpBarHandler.cs:13]: added (%) to exp bar, ergo when now it says "75/150 (50%)"

**Review by orenkoren — APPROVED** [review]

**Pull request merged**